File: iam-clojure/xlclient.clj
(use '[clojure.contrib.str-utils :only (str-join)])
(import '(java.util HashMap Iterator Properties)
        '(org.springframework.beans.factory.BeanCreationException)
        '(com.thortech.xl.util.config ConfigurationClient)
        '(oracle.iam.platform.OIMClient)
        '(oracle.iam.identity.usermgmt.api.UserManager)
        '(oracle.iam.identity.orgmgmt.vo.Organization)
        '(oracle.iam.identity.orgmgmt.api.OrganizationManager))

(defn login
  "Performs login and returns a tcUtilityFactory instance.
  There are three overloaded methods of this signature. The default
  method without any parameters will login using xelsysadm/xelsysadm.
  The method that takes a single parameter will attempt to perform
  signature based login with the userid provided. The method with
  the explicit login and password provided with use them to login"
  ([] (login "xelsysadm" "xelsysadm"))
  ([username password]
    (let [
        jndi (.getAllSettings (ConfigurationClient/getComplexSettingByPath "Discovery.CoreServer"))
        oimclient (oracle.iam.platform.OIMClient. jndi)
      ]
      (do 
        (.login oimclient username password)
        oimclient))))

(defn logout
  "Closes tcUtilityFactory instance passed"
  [oimclient]
  (do
    (println "Logging out")
    (.logout oimclient)))

File: iam-jython/xlclient.py
# OIM API interface

import java

from java.lang import *
from java.util import *
from oracle.iam.platform import *
import org.springframework.beans.factory.BeanCreationException
from oracle.iam.identity.usermgmt.api import *
from oracle.iam.identity.orgmgmt.api import *
from com.thortech.xl.util.config import *

class XLAPIClient:
    def __init__(self):
        self.intfMap = {
            'usrmgr' : Class.forName('oracle.iam.identity.usermgmt.api.UserManager'),
            'orgmgr' : Class.forName('oracle.iam.identity.orgmgmt.api.OrganizationManager'),
        }
        
        self.platform = None

    def defaultLogin(self):
        self.passwordLogin('xelsysadm', 'Welcome1')

    def passwordLogin(self, userID, password):
        try:
            print "Logging in"
            self.jndi = ConfigurationClient.getComplexSettingByPath('Discovery.CoreServer').getAllSettings()

            print "jndi = $jndi"

            self.platform = OIMClient(self.jndi)
            self.platform.login(userID, password)
        except Exception, ex:
            print "Exception occured ", ex.message
        
    def logout(self):
        print "Logging out"
        self.platform.logout()
        self.factory.close()

    def getUtility(self, which):
        clsName = self.intfMap[which]
        print str(clsName.__name__)
        self.platform.getService(clsName)
